src/utils/viewer/viewer.cpp

```diff
@@ -1093,7 +1164,23 @@ void Viewer::mousePressEvent(MouseEvent& event) {
     createPickedObjectVisualizer(pickedObject);
     return;
   }  // drawable selection
+  // Ctrl + Right click will visualize the voxel mesh of an object.
+  else if (event.button() == MouseEvent::Button::Right &&
+           event.modifiers() & MouseEvent::Modifier::Ctrl) {
+    if (simulator_->getPhysicsSimulationLibrary() !=
+        esp::physics::PhysicsManager::PhysicsSimulationLibrary::NONE) {
+      auto viewportPoint = event.position();
+      auto ray = renderCamera_->unproject(viewportPoint);
+      esp::physics::RaycastResults raycastResults = simulator_->castRay(ray);
 
+      if (raycastResults.hasHits()) {
+        auto objID = raycastResults.hits[0].objectId;
+#ifdef ESP_BUILD_WITH_VHACD
+        displayVoxelField(objID);
+#endif
+      }
+    }
+  }
```

Comment:
Can we merge this with the other right click functionality below to avoid code duplication? Consider checking for the ctrl key and diverging after line 1192.